synapse_data_warehouse/database_roles/V2.33.0__database_analyst_roles.sql

```diff
@@ -0,0 +1,96 @@
+    ON DATABASE ROLE SYNAPSE_RAW_ALL_ANALYST
+    TO ROLE SYNAPSE_RAW_ALL_ADMIN; --noqa: JJ01,PRS,TMP
+GRANT OWNERSHIP
+    ON DATABASE ROLE SCHEMACHANGE_ALL_ANALYST
+    TO ROLE SCHEMACHANGE_ALL_ADMIN; --noqa: JJ01,PRS,TMP
+
+-- Grant analyst aggregate roles to the database analyst account role
+GRANT DATABASE ROLE SYNAPSE_ALL_ANALYST
+    TO ROLE {{ database_name }}_ANALYST; --noqa: JJ01,PRS,TMP
+GRANT DATABASE ROLE SYNAPSE_RAW_ALL_ANALYST
+    TO ROLE {{ database_name }}_ANALYST; --noqa: JJ01,PRS,TMP
+GRANT DATABASE ROLE SCHEMACHANGE_ALL_ANALYST
+    TO ROLE {{ database_name }}_ANALYST; --noqa: JJ01,PRS,TMP
+
+
+-- Create object-type-specific roles, grant ownership of each to
+-- their respective admin aggregate role, and grant each to the
+-- respective analyst aggregate role
+-- SYNAPSE
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_TABLE_READ;
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_STAGE_READ;
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_VIEW_READ;
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_TASK_READ;
+
+GRANT OWNERSHIP
+	ON DATABASE ROLE SYNAPSE_TABLE_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ADMIN;
+GRANT OWNERSHIP
+	ON DATABASE ROLE SYNAPSE_STAGE_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ADMIN;
+GRANT OWNERSHIP
+	ON DATABASE ROLE SYNAPSE_VIEW_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ADMIN;
+GRANT OWNERSHIP
+	ON DATABASE ROLE SYNAPSE_TASK_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ADMIN;
+
+GRANT DATABASE ROLE SYNAPSE_TABLE_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ANALYST;
+GRANT DATABASE ROLE SYNAPSE_STAGE_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ANALYST;
+GRANT DATABASE ROLE SYNAPSE_VIEW_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ANALYST;
+GRANT DATABASE ROLE SYNAPSE_TASK_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ANALYST;
+
+-- SYNAPSE_RAW
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_RAW_TABLE_READ;
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_RAW_STAGE_READ;
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_RAW_STREAM_READ;
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_RAW_TASK_READ;
+
+GRANT OWNERSHIP
+	ON DATABASE ROLE SYNAPSE_RAW_TABLE_READ
+	TO DATABASE ROLE SYNAPSE_RAW_ALL_ADMIN;
+GRANT OWNERSHIP
+	ON DATABASE ROLE SYNAPSE_RAW_STAGE_READ
+	TO DATABASE ROLE SYNAPSE_RAW_ALL_ADMIN;
+GRANT OWNERSHIP
+	ON DATABASE ROLE SYNAPSE_RAW_STREAM_READ
+	TO DATABASE ROLE SYNAPSE_RAW_ALL_ADMIN;
+GRANT OWNERSHIP
+	ON DATABASE ROLE SYNAPSE_RAW_TASK_READ
+	TO DATABASE ROLE SYNAPSE_RAW_ALL_ADMIN;
+
+GRANT DATABASE ROLE SYNAPSE_RAW_TABLE_READ
+	TO DATABASE ROLE SYNAPSE_RAW_ALL_ANALYST;
+GRANT DATABASE ROLE SYNAPSE_RAW_STAGE_READ
+	TO DATABASE ROLE SYNAPSE_RAW_ALL_ANALYST;
+GRANT DATABASE ROLE SYNAPSE_RAW_STREAM_READ
+	TO DATABASE ROLE SYNAPSE_RAW_ALL_ANALYST;
+GRANT DATABASE ROLE SYNAPSE_RAW_TASK_READ
+	TO DATABASE ROLE SYNAPSE_RAW_ALL_ANALYST;
+
+-- SCHEMACHANGE
+CREATE OR REPLACE DATABASE ROLE SCHEMACHANGE_TABLE_READ;
+
+GRANT OWNERSHIP
+	ON DATABASE ROLE SCHEMACHANGE_TABLE_READ
+	TO DATABASE ROLE SCHEMACHANGE_ALL_ADMIN;
+
+GRANT DATABASE ROLE SCHEMACHANGE_TABLE_READ
+	TO DATABASE ROLE SCHEMACHANGE_ALL_ANALYST;
```

Comment:
I combined these steps into a single diagram to keep things simple. This is our final state.
![SNOW-200 (13)](https://github.com/user-attachments/assets/18d76fe2-b63f-4370-83b1-244064574c71)

Reply:
This diagram is super helpful and could be included in documentation (once final review) so that people have a easier time when onboarding and learning about this.

Reply:
I added the diagram [here](https://sagebionetworks.jira.com/wiki/spaces/DPE/pages/3829006353/Synapse+Data+Warehouse+Role+Hierarchy#Database-level). Turns out I already had some verbiage which said that all objects, including database roles, ought to be owned by the appropriate `*ALL_ADMIN` database role. But the diagram definitely makes it clearer.